### DISCUSSION TITLE
Black Screen and Disconnection Issue with Moonlight and Sunshine

### DISCUSSION THREAD

**Discussion by eliasstepanik** [LizardByte/community#416] [Sunshine] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

After setting up Sunshine on a Proxmox VM that runs Windows with GPU Passthrough, I attempted to connect via Moonlight from my main PC. Unfortunately, the connection results in a black screen, followed by disconnection. Shortly after, Moonlight displays the error message: "Received no video from host.".

### Expected Behavior

I expected Moonlight to successfully connect to the Sunshine server on the VM, allowing me to stream my desktop or games without issues.

### Additional Context

I have HDMI and DP Dongels connected to the GPU.
I also tested connecting a normal screen to the gpu and it showed the VM Windows Desktop.

### Host Operating System

Windows

### Operating System Version

Win 11

### Architecture

64 bit

### Sunshine commit or version

v0.22.2

### Package

Windows - installer

### GPU Type

AMD and NVIDA

### GPU Model

AMD Radeon RX 7900 XT, RTX 3070 ti, RTX 4080 Super

### GPU Driver/Mesa Version

24.3.1

### Config

```Shell
empty
```

### Relevant log output

```shell
[2024:03:30:14:18:15]: Info: Encoder reenumeration is required
[2024:03:30:14:18:15]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024:03:30:14:18:15]: Info: Trying encoder [nvenc]
[2024:03:30:14:18:16]: Info: 
Device Description : AMD Radeon RX 7900 XT
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000744C
Device Video Mem   : 20417 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 15981 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 1920x0
Virtual Desktop    : 3840x1080
[2024:03:30:14:18:16]: Info: Active GPU has HAGS enabled
[2024:03:30:14:18:16]: Info: Using realtime GPU priority
[2024:03:30:14:18:16]: Info: Desktop resolution [1920x1080]
[2024:03:30:14:18:16]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:03:30:14:18:16]: Info: Display refresh rate [60Hz]
[2024:03:30:14:18:16]: Info: Requested frame rate [60fps]
[2024:03:30:14:18:16]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.648438,0.338867]
Green Primary      : [0.282227,0.602539]
Blue Primary       : [0.142578,0.0703125]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:03:30:14:18:16]: Info: Encoder [nvenc] is not supported on this GPU
[2024:03:30:14:18:16]: Info: Trying encoder [quicksync]
[2024:03:30:14:18:18]: Info: 
Device Description : AMD Radeon RX 7900 XT
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000744C
Device Video Mem   : 20417 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 15981 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 1920x0
Virtual Desktop    : 3840x1080
[2024:03:30:14:18:18]: Info: Active GPU has HAGS enabled
[2024:03:30:14:18:18]: Info: Using realtime GPU priority
[2024:03:30:14:18:18]: Info: Desktop resolution [1920x1080]
[2024:03:30:14:18:18]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:03:30:14:18:18]: Info: Display refresh rate [60Hz]
[2024:03:30:14:18:18]: Info: Requested frame rate [60fps]
[2024:03:30:14:18:18]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.648438,0.338867]
Green Primary      : [0.282227,0.602539]
Blue Primary       : [0.142578,0.0703125]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:03:30:14:18:18]: Info: Encoder [quicksync] is not supported on this GPU
[2024:03:30:14:18:18]: Info: Trying encoder [amdvce]
[2024:03:30:14:18:20]: Info: 
Device Description : AMD Radeon RX 7900 XT
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000744C
Device Video Mem   : 20417 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 15981 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 1920x0
Virtual Desktop    : 3840x1080
[2024:03:30:14:18:20]: Info: Active GPU has HAGS enabled
[2024:03:30:14:18:20]: Info: Using realtime GPU priority
[2024:03:30:14:18:20]: Info: Desktop resolution [1920x1080]
[2024:03:30:14:18:20]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:03:30:14:18:20]: Info: Display refresh rate [60Hz]
[2024:03:30:14:18:20]: Info: Requested frame rate [60fps]
[2024:03:30:14:18:20]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.648438,0.338867]
Green Primary      : [0.282227,0.602539]
Blue Primary       : [0.142578,0.0703125]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:03:30:14:18:20]: Info: SDR color coding [Rec. 601]
[2024:03:30:14:18:20]: Info: Color depth: 8-bit
[2024:03:30:14:18:20]: Info: Color range: [JPEG]
[2024:03:30:14:18:21]: Info: SDR color coding [Rec. 601]
[2024:03:30:14:18:21]: Info: Color depth: 8-bit
[2024:03:30:14:18:21]: Info: Color range: [JPEG]
[2024:03:30:14:18:22]: Info: SDR color coding [Rec. 601]
[2024:03:30:14:18:22]: Info: Color depth: 8-bit
[2024:03:30:14:18:22]: Info: Color range: [JPEG]
[2024:03:30:14:18:24]: Info: 
Device Description : AMD Radeon RX 7900 XT
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000744C
Device Video Mem   : 20417 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 15981 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 1920x0
Virtual Desktop    : 3840x1080
[2024:03:30:14:18:24]: Info: Active GPU has HAGS enabled
[2024:03:30:14:18:24]: Info: Using realtime GPU priority
[2024:03:30:14:18:24]: Info: Desktop resolution [1920x1080]
[2024:03:30:14:18:24]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:03:30:14:18:24]: Info: Display refresh rate [60Hz]
[2024:03:30:14:18:24]: Info: Requested frame rate [60fps]
[2024:03:30:14:18:24]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.648438,0.338867]
Green Primary      : [0.282227,0.602539]
Blue Primary       : [0.142578,0.0703125]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:03:30:14:18:24]: Info: SDR color coding [Rec. 709]
[2024:03:30:14:18:24]: Info: Color depth: 10-bit
[2024:03:30:14:18:24]: Info: Color range: [JPEG]
[2024:03:30:14:18:24]: Info: SDR color coding [Rec. 709]
[2024:03:30:14:18:24]: Info: Color depth: 10-bit
[2024:03:30:14:18:24]: Info: Color range: [JPEG]
[2024:03:30:14:18:25]: Info: 
[2024:03:30:14:18:25]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024:03:30:14:18:25]: Info: 
[2024:03:30:14:18:25]: Info: Found H.264 encoder: h264_amf [amdvce]
[2024:03:30:14:18:25]: Info: Found HEVC encoder: hevc_amf [amdvce]
[2024:03:30:14:18:25]: Info: Found AV1 encoder: av1_amf [amdvce]
[2024:03:30:14:18:25]: Info: Executing [Desktop]
[2024:03:30:14:18:25]: Info: nvprefs: NvAPI_Initialize() failed, ignore if you don't have NVIDIA video card
[2024:03:30:14:18:25]: Info: CLIENT CONNECTED
[2024:03:30:14:18:27]: Info: 
Device Description : AMD Radeon RX 7900 XT
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000744C
Device Video Mem   : 20417 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 15981 MiB
Feature Level      : 0x0000B100
Capture size       : 1920x1080
Offset             : 0x0
Virtual Desktop    : 3840x1080
[2024:03:30:14:18:27]: Info: Active GPU has HAGS enabled
[2024:03:30:14:18:27]: Info: Using realtime GPU priority
[2024:03:30:14:18:27]: Info: Desktop resolution [1920x1080]
[2024:03:30:14:18:27]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:03:30:14:18:27]: Info: Display refresh rate [60Hz]
[2024:03:30:14:18:27]: Info: Requested frame rate [60fps]
[2024:03:30:14:18:27]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 8
Red Primary        : [0.648438,0.338867]
Green Primary      : [0.282227,0.602539]
Blue Primary       : [0.142578,0.0703125]
White Point        : [0.313477,0.329102]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:03:30:14:18:27]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:03:30:14:18:27]: Info: SDR color coding [Rec. 601]
[2024:03:30:14:18:27]: Info: Color depth: 8-bit
[2024:03:30:14:18:27]: Info: Color range: [MPEG]
[2024:03:30:14:18:36]: Info: CLIENT DISCONNECTED
```

**Comment by gschintgen** [comment]
So the project's maintainer closes your first report and you're just filing it again?

Anyway, you might want to give more details on the one thing that is non-standard: the virtualization aspect and in particular the networking. Is it bridged or routed? Did you check the sunshine docs for the relevant ports? Did you check if your client device running Moonlight can access those ports (notably video) on the sunshine VM? How did you check this?

**Reply by eliasstepanik** [reply]
I wrote a small Port testing tool. It seems that some TCP Ports are closed.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/13e480aa-6324-4129-81fd-41fe3538bc3e)

**Reply by eliasstepanik** [reply]
I also tested turning off the Vm's firewall in windows. Which didn´t help.

**Reply by eliasstepanik** [reply]
I tested turning off the firewall in Proxmox which I use for hosting the vm. Sadly that didn´t work ether.

**Reply by eliasstepanik** [reply]
What i found was that if I use the command `netstat -ano`

The 47990 Port ist constantly in a TIME_WAIT or CLOSE_WAIT State.

![image](https://github.com/LizardByte/Sunshine/assets/40958815/50a3cfd0-e003-428a-93e9-bb22726d8388)

**Reply by gschintgen** [reply]
That's weird indeed. I just ran nmap on the relevant ports on my sunshine host (nmap from a linux device to a linux sunshine host). All 4 TCP ports should indeed be open if sunshine is just sitting there waiting for a connection.
Maybe you could check on the sunshine host if it is even listening on those ports? Just to make sure. (On Linux I'd use netstat, don't know for Windows.) Apart from that I can't really help you. But at least you are one step closer to diagnosing the issue.
Out of interest (maybe also for other users): is your proxmox network bridged or routed?

(edit: I typed this comment before your netstat comment showed up.)

**Reply by eliasstepanik** [reply]
I use a Linux Bridge Network.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/81ee97ff-8da9-406d-acea-04567e7ffbfe)

**Reply by gschintgen** [reply]
re TIME_WAIT, CLOSE_WAIT: I'm not very knowledgeable about low level tcp internals, but I suppose you made sure that your do-it-yourself port tester properly closed or reset its connections?

**Reply by eliasstepanik** [reply]
I tested checking the Port on the Sunshine Host itself and it gives me the same output.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/47217cc3-ce0a-460b-bf78-aa12c4b4e34d)

**Reply by gschintgen** [reply]
Wait, that second column in `netstat -ano` is the "foreign address" i.e. the addresses for which it is listening or where it is connected to. Why does it say `::1` in there? Did you indeed connect on the sunshine host itself to sunshine? Or what does it mean. It should state `*:*` or `[::]:0` no?

**Reply by gschintgen** [reply]
I'd say you should reboot the VM and then run `netstat -ano` again.

**Reply by eliasstepanik** [reply]
I added the Proper disposal the the tester and it seems to think that they are all open now. I will try to verify with some other tool.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/3945546f-9ed1-41a9-93c6-810dd357fb22)

**Reply by gschintgen** [reply]
Hmm, I'm typing this right now on a Windows machine which has sunshine installed for testing purposes. It also has quite a lot of TIME_WAIT connections on ports 47984 and 47989. Those are all originating from the IP of my moonlight that is active right now. (I.e. loaded and showing the host selection screen). But why would you have such connections coming from `[:1]` i.e. localhost? You don't happen to have moonlight running on your sunshine vm? And nvidia's gamestream service is also properly disabled in the drivers?

**Reply by eliasstepanik** [reply]
I think that the Problem are the two UDP Port from below 48000 and 47998 because these two are not open.

Edit: This test is done on the Host itself. When I test it on my client I get the same result.

![image](https://github.com/LizardByte/Sunshine/assets/40958815/2d1f68e1-4f06-4138-adbf-031a2bd07f70)

**Reply by eliasstepanik** [reply]
It seems that the UDP Ports are sometimes open and sometimes closed. Which explaines why its not working correctly. What I can't seem to find is why its happening.

**Reply by ReenigneArcher** [reply]
The ports are opened when they need to be opened and closed otherwise.

**Reply by eliasstepanik** [reply]
That could be it but the problem what I found is when I test the udp ports, I always get the result Closed. Only after the 3 time the ports showes open.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/231c9e3b-a48c-4f19-aa98-b20ac1198689)

I could be that its just taking a bit of time to open them.

**Reply by eliasstepanik** [reply]
Gamestreaming is turned off.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/c39f80d9-45c4-4bfe-a04e-1109e714b5f0)

Moonlight is also not installed on the vm.

What I have is Parsec and Windows Remote Desktop but when I connect to sunshine I'm not using ether.

**Reply by eliasstepanik** [reply]
What I don't know is if GameStream is still enabled because its showing Gamestream (Ready) on the General Page und Features.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/b9c54363-9974-49e8-bfc0-e6a9e6906525)

**Comment by ReenigneArcher** [comment]
@gschintgen I closed it because they selected random items from the drop down lists.

![image](https://github.com/LizardByte/Sunshine/assets/42013603/714ba8f4-98cb-4eab-8994-b923d79514de)


Still, this is a pretty low quality bug report... and probably not really a bug.

**Comment by gschintgen** [comment]
Ah I see, the report has been edited later on. 

Anyway I'm vaguely interested in a VM setup myself, that's why I tried suggesting a few diagnostic steps. (But it does read more like a support request that could be moved to a discussion.)

**Comment by ReenigneArcher** [comment]
I'll leave it as an issue for now... if anything interesting comes out of it, perhaps the docs can be updated.

**Comment by eliasstepanik** [comment]
It seems to be a problem with the gpu software because when I tried it on windows running on bare metal I had the same issue. Sadly I can´t test more because I returned the gpus because they are far to expensive for them not to work the way I want them to.

After testing I think its not a gpu problem and more a network problem.

**Comment by eliasstepanik** [comment]
I found that this issue also accures when I use a different gpu this time a 3070 ti. I think its a network related issue. I checked all firewalls that could block this connection. But nothing fixes it.
![image](https://github.com/LizardByte/Sunshine/assets/40958815/588a595e-520e-4b9f-96f4-8616de5531b5)

**Comment by eliasstepanik** [comment]
I also think this Discussion could be related to the Problem.
https://github.com/LizardByte/Sunshine/discussions/1019

**Comment by eliasstepanik** [comment]
Ok I found something.
I tried installing sunshine on my main pc and tried connecting using my macbook. This worked without problems.
After that I tried using my Macbook to connect to my VM and for some strage reason everything is working.

**Reply by eliasstepanik** [reply]
I tested using a differnent Windows Maschine and it worked without any Problems. So the issue does is not on the vm. Its on my main PC.

**Comment by butch2k** [comment, marked as answer]
The issue is a strange one, and is linked to the proxmox network stack, i encountered it several times now. Reinstalling a fresh VM might solve it but it will happen again over time if you like me use VirtIO Paravirtualized Network Card (or Vmware one same issue). An update to Proxmox (or is it windows ?) might result in a non working Moonlight/Sunshine for... reasons... 

The only way to solve it is to change the network card to either an E1000, E1000E or RTL8139. You can do it in real time without stopping the VM and it solves the problem immediatly. If you go back to VirtIO it will break again.

**Reply by butch2k** [reply]
<img width="302" alt="image" src="https://github.com/user-attachments/assets/5f2fb33c-7f4b-456d-a95e-23ba6dd987d9" />
As a side note i was able to get sunshine/moonlight to work on VirtIo by manually setting the MTU Size to 1435, anything higher results in a black screen.

**Reply by gymnae** [reply]
That MTU tip was a god send. I reinstalled sunshine/apollo couple of times, reinstalled the GPU drivers... and no log filed helped me. This did. Thank you, @butch2k

**Reply by dizzydre21** [reply]
> <img alt="image" width="302" src="https://private-user-images.githubusercontent.com/961507/443417533-5f2fb33c-7f4b-456d-a95e-23ba6dd987d9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgwMTQ4MzAsIm5iZiI6MTc0ODAxNDUzMCwicGF0aCI6Ii85NjE1MDcvNDQzNDE3NTMzLTVmMmZiMzNjLTdmNGItNDU2ZC1hOTVlLTIzYmE2ZGQ5ODdkOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyM1QxNTM1MzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xM2YzZDcyZGEyNmZmN2Y3MDI5Y2Q3ODI0MDI0YjY1ODgxYzI3MzQyOGQ1ZWYyYzgxYjE5OWNkNDA5MmE2ZWU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.7DxRUcfQuklGzt_3z60G-IrSANflvfqSqnIrXYdwnh0"> As a side note i was able to get sunshine/moonlight to work on VirtIo by manually setting the MTU Size to 1435, anything higher results in a black screen.

I am unable to get virtio to work even with the MTU set. E1000 does work though.

I'll also give a bit more detail. Windows client devices have no issues with virtio. It's only android or Linux clients for me. A fresh Windows 11 Pro install allows me to use the Android and Linux clients for a short time. I have a sneaking suspicion that a Windows update is happening and screwing it up somehow. I tried removing the updates that occur shortly after the first boot a fresh install, but it doesn't let me uninstall all of them.

**Reply by jerrin-f** [reply]
As with @dizzydre21, I couldn't get the virtio adapter working with MTU 1435 either, but disabling UDP Segmentation Offload via `Device Manager > Network Adapters > VirtIO Adapter > Properties > Advanced` did the trick.

**Reply by dizzydre21** [reply]
> As with @dizzydre21, I couldn't get the virtio adapter working with MTU 1435 either, but disabling UDP Segmentation Offload via `Device Manager > Network Adapters > VirtIO Adapter > Properties > Advanced` did the trick.

Interesting. I briefly read up on that after your post, but it's beyond my networking knowledge. What, if any, penalties occur when disabling it?

**Reply by DarkLegacy248** [reply]
Oh, my gosh, thank you so much.  I'm still new to proxmox, moonlight and sunshine.  Still making the transition from VMware.  Changing the MTU setting to 1435 worked like a charm.  Narrowed it down with being a slight change from Pfsense 2.7.2 to 2.8.0 somehow killed it all working with standard MTU 1500.  If will have to investigate what specially changed.  I don't know how you figured this out but I'm so grateful you did.  Thank you.

**Reply by butch2k** [reply]
> As with @dizzydre21, I couldn't get the virtio adapter working with MTU 1435 either, but disabling UDP Segmentation Offload via `Device Manager > Network Adapters > VirtIO Adapter > Properties > Advanced` did the trick.

Yeah i noticed it too, there is definitely an issue with packet size/fragmentation with paravirtualization and i just can't understand what exactly is causing this. What proxmox kernel version are u using ? the 6.8 or 6.11 ? 

UDP offloading should not change much in performance, with a real network card it's used to get the network card do what the CPU does but in paravirtualized networking i believe it just pass the stuff to one of the cores used by proxmox itself, not the underlaying real network card.

**Reply by Kraust** [reply]
> The issue is a strange one, and is linked to the proxmox network stack, i encountered it several times now. Reinstalling a fresh VM might solve it but it will happen again over time if you like me use VirtIO Paravirtualized Network Card (or Vmware one same issue). An update to Proxmox (or is it windows ?) might result in a non working Moonlight/Sunshine for... reasons...
> 
> The only way to solve it is to change the network card to either an E1000, E1000E or RTL8139. You can do it in real time without stopping the VM and it solves the problem immediatly. If you go back to VirtIO it will break again.

My guy, I have spent the past EIGHT hours debugging this issue, including blowing away my entire operating system and reinstalling Proxmox instead of NixOS running Proxmox to find your comment. All of a sudden this morning Moonlight refused to connect to my Windows VM, I can access said Windows VM through Moonlight from another VM but not my host. Switching to E1000 fixes it for me. This is absolutely baffling and must have taken you ages to debug. I will have to figure out how to get Virtio working at some point because this will drive me crazy until I do. Will try some of these other fixes next.

----

Setting the MTU resolved it for me.

**Reply by eliasstepanik** [reply]
I fixed the problem by switching the network model but since I stopped using proxmox for my windows vms because its overall instability a while ago I don‘t exactly remember. 


I just know that it was a pain to get fixed and even then it was not even remotely stable. It worked most of the time but sometimes wired connection issues and random disconnects happen. I just ditched proxmox and moved back to running windows barebones on my server.

**Reply by dizzydre21** [reply]
> The issue is a strange one, and is linked to the proxmox network stack, i encountered it several times now. Reinstalling a fresh VM might solve it but it will happen again over time if you like me use VirtIO Paravirtualized Network Card (or Vmware one same issue). An update to Proxmox (or is it windows ?) might result in a non working Moonlight/Sunshine for... reasons...
> 
> The only way to solve it is to change the network card to either an E1000, E1000E or RTL8139. You can do it in real time without stopping the VM and it solves the problem immediatly. If you go back to VirtIO it will break again.

Hey, have you had any movement on this issue?

I have a similar issue that I'm thinking may be related when using 10gb NICs and virtio in Proxmox and Moonlight/Sunshine. I've got a new machine set up solely for two gaming VMs, one for Linux and one for Win11. With both Broadcom and Intel 10gb NICs, I get an unusable amount of stutter and frame drops, but it only happens when using virtio as the adapter. I can RDP into the VM, though, and it works without issue. I can even transfer a large video file to the VM at 500MBps, which is as fast as my NAS pool can transfer. Changing to either E1000 adapter works without issue. I can pull the 10Gb NIC out, install a 2.5Gb Realtek, and there are zero problems using virtio as long as MTU is st to 1435.

**Reply by nodegin** [reply]
@butch2k @jerrin-f Thank you bros, setting both fixed the black screen on connection issue for me. It started to happen when I upgraded the Windows 11 in the VM. You saved my life!

**Reply by abla7e** [reply]
Thank you SO MUCH. What an obscure bug, incredible that someone has the exact solution written out.

**Comment by Deceased-prince** [comment]
i found that passing through a physical nic will fix this or changing the model
